es/utils/swagger.js

```diff
@@ -16,513 +16,82 @@
-          } else {
-            throw Error(`Function call doesn't conform to ${signature}`)
-          }
-        })()
-
-        if (opt.debug) {
-          console.log(`Invoked ${name} with ${R.toString(arg)} ${R.toString(opt)}`)
-        }
-
-        const values = R.merge(R.reject(R.isNil, R.pick(optNames, opt)), R.zipObj(R.pluck('name', req), arg))
-        const conformed = R.mapObjIndexed((val, key) => {
-          try {
-            return conform(val, indexedParameters[key], swag.definitions)
-          } catch (e) {
-            const path = [key].concat(e.path || [])
-            throw Object.assign(e, {
-              path,
-              value: val,
-              message: `validating ${R.join(' -> ', path)}: ${e.message}`
-            })
-          }
-        }, values)
-        const expandedPath = expandPath(path, snakizeKeys(R.pick(pathArgs, conformed)))
-        const params = snakizeKeys((() => {
-          if (method === 'get') {
-            return { params: R.pick(queryArgs, conformed) }
-          } else if (method === 'post') {
-            return conformed[assertOne(bodyArgs)]
-          } else {
-            throw Error(`Unsupported method ${method}`)
-          }
-        })())
-
-        if (opt.debug) {
-          console.log(`Going to ${method.toUpperCase()} ${url}${expandedPath} with ${R.toString(params)}`)
-        }
-
-        try {
-          const response = await client(`${url}${expandedPath}`, params).catch(this.axiosError(errorHandler))
-          // return opt.fullResponse ? response : conform(pascalizeKeys(response.data), responses['200'], swag.definitions)
-          return opt.fullResponse ? response : pascalizeKeys(response.data)
-        } catch (e) {
-          if (R.path(['response', 'data'], e)) {
-            e.message = destructureClientError(e)
-          }
-          throw e
-        }
-      } catch (e) {
-        e.message = `While calling ${signature}, ${e.message}`
-        throw e
-      }
-    }.bind(instance)
-
-    Object.assign(fn, {
-      signature,
-      description
-    })
-
-    return Object.defineProperties(fn, {
-      name: {
-        value: name,
-        writable: false
-      },
-      length: {
-        value: req.length + (opts.length ? 1 : 0),
-        writable: false
+ * @alias module:@aeternity/aepp-sdk/es/utils/swagger
+ * @rtype Object
+ * @param {String} specUrl - Swagger specification URL on external node host
+ * @param {String} internalUrl - Node internal URL
+ * @return {Object} Swagger client
+ * @example (await genSwaggerClient('https://mainnet.aeternity.io/api')).getAccountByPubkey('ak_jupBUgZNbcC4krDLR3tAkw1iBZoBbkNeShAq4atBtpFWmz36r')
+ */
+export default async (specUrl, internalUrl) => {
+  const spec = await (await fetch(specUrl)).json();
+
+  ['claim', 'preclaim', 'transfer', 'revoke', 'update'].forEach(name => {
+    const s = spec.paths[`/debug/names/${name}`]?.post
+    if (s && !s.consumes) s.consumes = ['application/json']
+  })
```

Comment:
this should be removed after resolving https://github.com/aeternity/aeternity/issues/3539

Reply:
Can you log this into an issue, or mark this PR as `wip` until that is fixed?

Reply:
I have created #1124